src/qtism/runtime/tests/AbstractSessionManager.php

```diff
@@ -200,6 +202,7 @@ protected function createRoute(AssessmentTest $test): Route
                             // Do the same as for branch rules for pre conditions, except that they must be
                             // attached on the first item of the route.
                             $route->getFirstRouteItem()->addPreConditions($current->getPreConditions());
+                            $route->getFirstRouteItem()->addPreConditions($testPart->getPreConditions());
```

Comment:
For this one I used a bit different approach now, so the routeItem retains the knowledge of which pre-conditions to use as it needs to respect testPart/section/item pre-conditions. 

I've added it here https://github.com/oat-sa/qti-sdk/pull/367/files#diff-f33df5e2727b0618d5b89c14e69336ff58245c1453d97a797446bc2336d38cceR276

Reply:
I've also fixed an issue that allowed an item to be accessed in case 1 of the preconditions are met instead of all preconditions are met. 

👉 https://github.com/oat-sa/qti-sdk/pull/367/commits/85f9d100ef6e630175ec57053d2bfba81b59a21f

cc: @julien-sebire